pkg/util/refreshable/refreshable.go

```diff
@@ -61,7 +61,7 @@ func (a *authorizer) RefreshWithContext(ctx context.Context, log *logrus.Entry)
 		if !autorest.IsTokenRefreshError(err) ||
 			azureerrors.IsUnauthorizedClientError(err) ||
 			azureerrors.IsInvalidSecretError(err) {
-			return false, nil
+			return false, err
```

Comment:
Returning `false, err` is the same as what is being returned after the `if` statement block on line 67, which means the `if` statement isn't actually required to get the same return values from the function. However, the check is here for a reason, as detailed in the comments above the function, we need to retry certain failures to overcome AAD propagation delays.
So, while this change does work and the Invalid Service Principal error is returned it looks like we've lost retries as a result. Is that correct?

Reply:
Yes it is being same as the return statement outside the inner if. If we require the retries then we can return false, nil itself here and can make the changes in the file: /pkg/util/aad/aad.go#L48 by adding following code fragment:
if !done  && err == nil {
err = api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidServicePrincipalCredentials, "properties.servicePrincipalProfile", "The provided secret is invalid. or Unauthorized client or Token Error")

// or any new errors 
}

is it fine?